app/src/main/java/ee/mcdimus/androidacademyapp/MainActivity.kt

```diff
@@ -1,25 +1,35 @@
 package ee.mcdimus.androidacademyapp
 
-import android.content.Intent
 import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
-import android.widget.Button
 
-class MainActivity : AppCompatActivity() {
+class MainActivity : AppCompatActivity(), FragmentNavigationListener {
+
+  private val rootFragment = FragmentMoviesList()
 
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
     setContentView(R.layout.activity_main)
 
-    val pushButton = findViewById<Button>(R.id.btn_open_movie_details)
-    pushButton.setOnClickListener {
-      startMovieDetailsActivity()
+    if (savedInstanceState == null) {
+      supportFragmentManager.beginTransaction().apply {
+        add(R.id.main_content, rootFragment)
+        commit()
+      }
```

Comment:
Можно обойтись без apply, api позволяет. И сконструировать фрагмент при добавлении.
```suggestion
      supportFragmentManager.beginTransaction()
          .add(R.id.main_content, FragmentMoviesList())
          commit()
```